scripts/signers_framework/signers_analysis.py

```diff
@@ -0,0 +1,102 @@
+from datetime import datetime, timezone
+import requests
+import pandas as pd
+from rich.console import Console
+from rich.pretty import pprint
+
+from brownie import Contract, web3
+from helpers.addresses import r
+
+console = Console()
+
+# NOTE: request info and fill it up over time depending on rotations
+replacements = {
+    "techops_multisig": {
+        "0x10C7a2ca16116E5C998Bfa3BC9CEF464475B18ff": "0x0a9af7FAba0d5DF7A8C881e1B9cd679ee07Af8A2",
+        "0xeE8b29AA52dD5fF2559da2C50b1887ADee257556": "0xfA5bb45895Cb3C0aE5B1583Fe068f009A48F0187",
+        "0xBB2281cA5B4d07263112604D1F182AD0Ab26a252": "0xE8eA1D8B3a5A4CEC7E94AE330fF18E82B5D22fA6",
+    }
+}
+
+# we introduce times of judgement as args (default: Q2)
+# likely worth running once at the end of each quarter
+def main(start_time=1648771200, end_time=1656547200):
+    msigs = r.badger_wallets
+
+    console.print(
+        f"Time interval inspected: [{datetime.fromtimestamp(start_time, tz=timezone.utc)}, {datetime.fromtimestamp(end_time, tz=timezone.utc)}]"
+    )
+
+    for key, addr in msigs.items():
+        if "multisig" in key:
+            console.print(f"[green] Inspecting {key} at {addr}[/green]")
+
+            safe = Contract(addr)
+
+            total_tx = 0
+
+            # judging signed amount
+            owners_signed = dict.fromkeys(safe.getOwners(), 0)
+
+            url = f"https://safe-transaction.gnosis.io/api/v1/safes/{web3.toChecksumAddress(addr)}/transactions/"
+            urls = [url]
+
+            for next_url in urls:
```

Comment:
i dont understand what is happening here. in what case is `len(urls) != 1`?

Reply:
was just to facilitate the `next_url` in the flight, but the length of urls never will be more than one

Reply:
removed loop in [3efc70a](https://github.com/Badger-Finance/badger-multisig/pull/749/commits/3efc70a7761b4dfae4a2d3e3c73f6a60475e620d)